ext/reflection/php_reflection.c

```diff
@@ -5942,6 +6023,47 @@ ZEND_METHOD(ReflectionProperty, getType)
 }
 /* }}} */
 
+ZEND_METHOD(ReflectionProperty, getSettableType)
+{
+	reflection_object *intern;
+	property_reference *ref;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	GET_REFLECTION_OBJECT_PTR(ref);
+
+	zend_property_info *prop = ref->prop;
+	/* Dynamic property is untyped. */
+	if (!ref->prop) {
+		RETURN_NULL();
+	}
+
+	/* Get-only virtual property can never be written to. */
+	if ((prop->flags & ZEND_ACC_VIRTUAL) && !prop->hooks[ZEND_PROPERTY_HOOK_SET]) {
```

Comment:
Is it the intention that ZEND_ACC_VIRTUAL implies the existence of hooks? If we allow to specify `@virtual` in the stub file in a follow-up to avoid allocating backing storage, then this check can potentially dereference a NULL pointer because prop->hooks isn't necessarily filled in.

Reply:
I don't know all the checks by heart but at least here, `ZEND_ACC_VIRTUAL` assumes hooks are present. I presume your intention is to use `@virtual` for existing virtual properties implemented using the existing object property handlers? Tweaking the logic where necessary to check for the existence of hooks would be ok with me, if these cannot immediately be migrated to hooks (in part because internal classes aren't supported yet). But I'd prefer doing that after this has merged.

Reply:
> I presume your intention is to use @virtual for existing virtual properties implemented using the existing object property handlers?

Indeed.

> Tweaking the logic where necessary to check for the existence of hooks would be ok with me, if these cannot immediately be migrated to hooks (in part because internal classes aren't supported yet)

I don't know how internal hooks would look like so can't really comment on this (yet).

> But I'd prefer doing that after this has merged.

Of course

Reply:
> I don't know how internal hooks would look like so can't really comment on this (yet).

My intention is to allow marking the property with `{ get; set; }` (or possibly just annotations until we have PHP-Parser support), which will look and link the corresponding function into `zend_property_info.hooks`. So, hopefully it will not be too complicated.

Reply:
Ah, those would be zend_function's right? That's a bit of a pitty because it would require completely reworking the virtual properties in DOM/XMLReader with the additional penalty of having to deal with a stackframe, unless they can be frameless.

Reply:
:thinking: Good question. Avoiding the stack frame was not planned for now, but I'll see what's possible once I get to it.